app/controllers/concerns/two_factor_authenticatable_methods.rb

```diff
@@ -168,9 +168,9 @@ def update_invalid_user
     current_user.increment_second_factor_attempts_count!
   end
 
-  def handle_valid_verification_for_confirmation_context
-    user_session[:authn_at] = Time.zone.now
-    @next_mfa_setup_path = next_setup_path
```

Comment:
This line is only used by the `otp_verification_controller` when in a confirmation context, so it was moved there. Ideally I would have moved it in #8424